tempodb/search/streaming_search_block.go

```diff
@@ -112,15 +114,20 @@ func (s *StreamingSearchBlock) Search(ctx context.Context, p Pipeline, sr *Resul
 		}
 		buf = buf[:r.Length]
 
-		_, err := s.file.ReadAt(buf, int64(r.Start))
+		pagesBuffer, buffer, err = dr.Read(ctx, []common.Record{r}, pagesBuffer, buffer)
```

Comment:
This looks like similar functionality to `StreamingSearchBlock.Iterator()`, could we reuse the iterator here?  One downside might be buffer reuse, as the iterator returns a new buffer each time.